jax/interpreters/ad.py

```diff
@@ -556,10 +576,22 @@ def call_transpose(primitive, params, jaxpr, consts, freevar_vals, args, ct):
 
 def map_transpose(primitive, params, jaxpr, consts, freevar_vals, args, ct):
   all_args, in_tree_def = tree_flatten((consts, freevar_vals, args, ct))
+  # I think only one of `args` is non-undefined, but is only one of `ct` non-Zero?
+  # if that were the case we could use integers instead of tuples for the last two
+  # as is, it's unclear if it's even possible to create tree prefixes...
+  # instead, we'll want to separately flatten consts and freevars (using flatten_axes)
+  # and the other two
+  primal_dim = tuple(dim for arg, dim in zip(args, params['in_dims']) if arg is not undefined_primal)
+  in_dims = (0,) * len(tree_flatten((consts, freevar_vals))[0]) + \
```

Comment:
All this will need to be revisited in light of the fact that `consts` and `freevar_vals` are now part of `args` :-(

Reply:
There is also params['mapped_invars'] that tells you which `args` are mapped over.